src/app/components/client/LocationAutocompleteInput.tsx

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ChangeEvent, useEffect, useDeferredValue, useState } from "react";
+import {
+  TMatchingLocations,
+  ISearchLocationParams,
+  ISearchLocationResults,
+} from "../../api/v1/location-autocomplete/route";
+
+const getLocationSuggestions = async ({
+  searchParams,
+  abortController,
+}: {
+  searchParams: ISearchLocationParams;
+  abortController: AbortController;
+}): Promise<ISearchLocationResults | null> => {
+  try {
+    const { signal } = abortController;
+    const response = await fetch("/api/v1/location-autocomplete", {
+      method: "POST",
+      body: JSON.stringify(searchParams),
+      signal,
+    });
+
+    if (!response.ok) {
+      return null;
+    }
+
+    const locationResults = await response.json();
+    return locationResults;
+  } catch (_) {
+    return null;
+  }
+};
+
+export const LocationAutocompleteInput = () => {
+  const [locationSuggestions, setLocationSuggestions] =
+    useState<TMatchingLocations>([]);
+  const [searchQuery, setSearchQuery] = useState("");
+  const deferredSearchQuery = useDeferredValue(searchQuery);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    if (deferredSearchQuery) {
+      const searchParams = {
+        searchQuery: deferredSearchQuery,
+        config: {
+          minQueryLength: 1,
+          maxResults: 5,
+        },
+      };
+
+      getLocationSuggestions({
+        searchParams,
+        abortController: abortController,
+      }).then((data) => {
+        if (data) {
+          setLocationSuggestions(data.results);
+        }
+      });
+    } else {
+      setLocationSuggestions([]);
+    }
+
+    return () => {
+      abortController.abort();
```

Comment:
Just verifying that this appears to work well; if a new request is sent while the previous response hasn't come in yet, it'll abort the request.

![image](https://github.com/mozilla/blurts-server/assets/4251/45b89645-e19a-4d78-9821-4527a1547e9c)